src/v/cloud_storage/cache_service.cc

```diff
@@ -796,6 +804,43 @@ ss::future<> cache::do_reserve_space(size_t bytes) {
     _reserved_cache_size += bytes;
 }
 
+void cache::set_block_puts(bool block_puts) {
+    if (_block_puts && !block_puts) {
+        _block_puts_cond.signal();
+    }
+    _block_puts = block_puts;
+}
+
+void cache::notify_disk_status(
+  [[maybe_unused]] uint64_t total_space,
+  [[maybe_unused]] uint64_t free_space,
+  storage::disk_space_alert alert) {
+    vassert(ss::this_shard_id() == 0, "Called on wrong shard");
+
+    bool block_puts = (alert == storage::disk_space_alert::degraded);
+
+    // Log on transitions
+    if (block_puts && !_block_puts) {
+        vlog(
+          cst_log.warn,
+          "Tiered storage cache blocking segment promotions, disk space is "
+          "critically low.");
+    } else if (!block_puts && _block_puts) {
+        vlog(
+          cst_log.info,
+          "Tiered storage cache un-blocking promotions, disk space is no "
+          "longer critical.");
+    }
+
+    // Propagate to other shards on changes
+    if (block_puts != _block_puts) {
```

Comment:
nit: maybe move this up into the other check? unless we plan on adding anything else after between logging and setting the blocks

Reply:
done